wire-ios-sync-engine/Tests/Source/SessionManager/SessionManagerHelper.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+@testable import WireSyncEngine
+
+struct SessionManagerHelper {
```

Comment:
Just an opinion:
This is actually a factory, isn't it? It could even be in the production target, in my opinion.
It could look like this:
```
public struct SessionManagerFactory {

    public var maxNumberAccounts: Int
    public var jailbreakDetector: JailbreakDetectorProtocol
    public var dispatchGroup: ZMSDispatchGroup
    // ... the other properties like mediaManager, flowManager whatever

    public init(
        maxNumberAccounts: Int = SessionManager.defaultMaxNumberAccounts,
        jailbreakDetector: JailbreakDetectorProtocol = JailbreakDetector(),
        dispatchGroup: ZMSDispatchGroup = ZMSDispatchGroup()
        // ...
    ) {
        self.maxNumberAccounts = maxNumberAccounts
        self.jailbreakDetector = jailbreakDetector
        self.dispatchGroup = dispatchGroup
        // ...
    }

    public func createSessionManager() -> SessionManager {
        // ...
    }
}
```
In the test target a convenience initialiser could set all the properties to mocks and in the tests, where it is actually needed, it could look like that:
```
        var factory = SessionManagerFactory.allMocked
        factory.dispatchGroup = dispatchGroup
        let sut = factory.createSessionManager()
```

Reply:
This is a nice suggestion. As it is for test only I prefer to have it in a test target. I named it `SessionManagerBuilder` for now, because it is more complex than a normal factory to setup the object composition.